src/Risk.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Risk where
+
+import Control.Monad
+import Data.List
+import Control.Monad.Random
+
+------------------------------------------------------------
+-- Die values
+
+newtype DieValue = DV { unDV :: Int }
+  deriving (Eq, Ord, Show, Num)
+
+first :: (a -> b) -> (a, c) -> (b, c)
+first f (a, c) = (f a, c)
+
+instance Random DieValue where
+  random           = first DV . randomR (1,6)
+  randomR (low,hi) = first DV . randomR (max 1 (unDV low), min 6 (unDV hi))
+
+die :: Rand StdGen DieValue
+die = getRandom
+
+------------------------------------------------------------
+-- Risk
+
+type Army = Int
+
+data Battlefield = Battlefield { attackers :: Army, defenders :: Army }
+  deriving Show
+
+-----------------------------------------------------------------
+-- Exercise 2: battle
+-----------------------------------------------------------------
+-- sorts the dice rolls of a team into descending order
+diceSort :: Int -> Rand StdGen [Int]
+diceSort n = (map unDV) . (reverse . sort) <$> (replicateM n die)
```

Comment:
If you want to sort in descending order, it's better to use `sortBy (flip compare)` instead of doing `reverse`. Using `reverse` is less efficient.